tests/tools/test_dist_part.py

```diff
@@ -140,31 +61,65 @@ def rand_edges(num_src, num_dst, num_edges):
                 assert feat_array.shape[0] == num_papers // num_chunks
 
         # check edge_data
+        edge_data_gold = {}
         num_edges = {
             'paper:cites:paper': num_cite_edges,
             'author:writes:paper': num_write_edges,
-            'paper:rev_writes:author': num_write_edges
+            'paper:rev_writes:author': num_write_edges,
         }
         output_edge_data_dir = os.path.join(output_dir, 'edge_data')
         for etype, feat in [
             ['paper:cites:paper', 'count'],
             ['author:writes:paper', 'year'],
-            ['paper:rev_writes:author', 'year']
+            ['paper:rev_writes:author', 'year'],
         ]:
             output_edge_sub_dir = os.path.join(output_edge_data_dir, etype)
+            features = []
             for i in range(num_chunks):
                 chunk_f_name = '{}-{}.npy'.format(feat, i)
                 chunk_f_name = os.path.join(output_edge_sub_dir, chunk_f_name)
                 assert os.path.isfile(chunk_f_name)
                 feat_array = np.load(chunk_f_name)
-                assert feat_array.shape[0] == num_edges[etype] // num_chunks
+            assert feat_array.shape[0] == num_edges[etype] // num_chunks
+            features.append(feat_array)
+            edge_data_gold[etype + '/' + feat] = np.concatenate(features)
```

Comment:
do we need to save edge features in this testcase? seem not needed.

Reply:
This is one way to test the data or we can use the graph object as well. But I prefer reading it from files which gives us to indications about the test (that the data was properly stored in the graph which it was being generated.